### PR TITLE
BUG: fix s3 glob

### LINKED ISSUE #607: BUG:pd.read_parquet from s3 path using '*' run into 'IndexError: string index out of range'
### Describe the bug
read parquet from s3 using '*' run into 'IndexError: string index out of range'

### To Reproduce
To help us to reproduce this bug, please provide information below:

1. Your Python version: 3.9.12
2. The version of Xorbits you use: 0.4.2
3. Versions of crucial packages, such as numpy, scipy and pandas: pandas==1.4.2
4. Full stack of the error.
5. Minimized code to reproduce the error.

Code 
```python
import os
import xorbits.pandas as pd

option = {
    "key": "****",
    "secret": "****",
    "endpoint_url": "https://cos.ap-beijing.myqcloud.com",
}

df = pd.read_parquet(
    "s3://<bucket>/<path>/pt=20230403/*parquet",
    storage_options=option,
)
print(df.head())
```
Error message:
```text
Traceback (most recent call last):
  File "/Users/***/workspace/xorbits/test_s3.py", line 20, in <module>
    df = pd.read_parquet(
  File "/Users/***/lib/anaconda3/lib/python3.9/site-packages/xorbits/core/adapter.py", line 472, in wrapped
    return from_mars(c(*to_mars(args), **to_mars(kwargs)))
  File "/Users/***/lib/anaconda3/lib/python3.9/site-packages/xorbits/_mars/dataframe/datasource/read_parquet.py", line 772, in read_parquet
    file_path = glob(path, storage_options=storage_options)[0]
  File "/Users/***/lib/anaconda3/lib/python3.9/site-packages/xorbits/_mars/lib/filesystem/core.py", line 77, in glob
    return fs.glob(path)
  File "/Users/***/lib/anaconda3/lib/python3.9/site-packages/xorbits/_mars/lib/filesystem/fsspec_adapter.py", line 94, in glob
    FileSystemGlob(self).glob(self._normalize_path(path), recursive=recursive)
  File "/Users/***/lib/anaconda3/lib/python3.9/site-packages/xorbits/_mars/lib/filesystem/_glob.py", line 60, in glob
    return list(self.iglob(pathname, recursive=recursive))
  File "/Users/***/lib/anaconda3/lib/python3.9/site-packages/xorbits/_mars/lib/filesystem/_glob.py", line 112, in _iglob
    for name in glob_in_dir(dirname, basename, dironly):
  File "/Users/***/lib/anaconda3/lib/python3.9/site-packages/xorbits/_mars/lib/filesystem/_glob.py", line 126, in _glob1
    return fnmatch.filter(names, pattern)
  File "/Users/***/lib/anaconda3/lib/python3.9/fnmatch.py", line 61, in filter
    for name in names:
  File "/Users/***/lib/anaconda3/lib/python3.9/site-packages/xorbits/_mars/lib/filesystem/_glob.py", line 125, in <genexpr>
    names = (x for x in names if not _ishidden(x))
  File "/Users/***/lib/anaconda3/lib/python3.9/site-packages/xorbits/_mars/lib/filesystem/_glob.py", line 35, in _ishidden
    return path[0] in (".", b"."[0])
IndexError: string index out of range
```

### PR DESCRIPTION
**Pull request by Hank0626**
## Related issue number

Fixes #607 

## Check code requirements

- [x] tests added / passed (if needed)
- [x] Ensure all linting tests pass

### CODE REVIEW

**Review by aresnow1 — APPROVED** [review]
LGTM

File: python/xorbits/_mars/lib/filesystem/_glob.py [hunk 1]
```diff
@@ -157,11 +157,14 @@ def _glob2(self, dirname, pattern, dironly):  # pragma: no cover
     def _iterdir(self, dirname, dironly):
         if not dirname:  # pragma: no cover
             dirname = ""
-        if not self._fs.isdir(dirname):
+        if self._fs.isfile(dirname):
+            return iter([])
+        try:  # pragma: no cover
```
Comment (UranusSeven) [inline python/xorbits/_mars/lib/filesystem/_glob.py:162]: Could you explain a little bit about the try and except? Like when will an exception be raised?

Comment (Hank0626) [reply python/xorbits/_mars/lib/filesystem/_glob.py:162]: when the dirname does not exist in the fs, it will just return an empty iterator.

**Review by UranusSeven — APPROVED** [review]

**Review by aresnow1 — APPROVED** [review]
LGTM

**Pull request merged**